### PR TITLE
letsencrypt: support Google Domains DNS challenge

### PR DESCRIPTION
**Pull request by jbms**
Fixes https://github.com/home-assistant/addons/issues/834

### CODE REVIEW

**Comment by iAbadia** [comment]
LGTM!

File: letsencrypt/rootfs/etc/cont-init.d/file-structure.sh [hunk 1]
```diff
@@ -46,7 +46,12 @@ echo -e "dns_cloudxns_api_key = $(bashio::config 'dns.cloudxns_api_key')\n" \
       "dns_inwx_url = https://api.domrobot.com/xmlrpc/\n" \
       "dns_inwx_username = $(bashio::config 'dns.inwx_username')\n" \
       "dns_inwx_password = $(bashio::config 'dns.inwx_password')\n" \
-      "dns_inwx_shared_secret = $(bashio::config 'dns.inwx_shared_secret')" > /data/dnsapikey
+      "dns_inwx_shared_secret = $(bashio::config 'dns.inwx_shared_secret')\n" \
+      "dns_google_domains_access_token = $(bashio::config 'dns.google_domains_access_token')\n" > /data/dnsapikey
+
+if bashio::config.exists 'dns.google_domains_zone'; then
+      echo -e "dns_google_domains_zone = $(bashio::config 'dns.google_domains_zone')\n" >> /data/dnsapikey
+fi
```
Comment (agners) [inline letsencrypt/rootfs/etc/cont-init.d/file-structure.sh:54]: The `dns.google_domains_zone` config is optional just like many others, why do we need to special case it here? Can we not just assign it no matter whether the config is set or not?

Comment (jbms) [reply letsencrypt/rootfs/etc/cont-init.d/file-structure.sh:54]: I think the plugin does not treat unset the same as set to an empty string.

Comment (agners) [reply letsencrypt/rootfs/etc/cont-init.d/file-structure.sh:54]: Hm, can you double check if that really is the case? Seems weird to me that all the other plug-ins don't need that special casing...

**Review by agners — APPROVED** [review]

File: letsencrypt/DOCS.md [hunk 2]
```diff
@@ -210,6 +212,30 @@ on the DNS zone to be used for authentication.
...
+  challenge: dns
+  dns:
+    provider: dns-google-domains
+    google_domains_access_token: XXXX
+    google_domains_zone: home-assistant.io
+  ```
+
+  To obtain the ACME DNS API token follow the instructions here:
+
+  <https://support.google.com/domains/answer/7630973?hl=en#zippy=%2Clist-of-known-acme-clients-to-work-the-acme-dns-api>
```
Comment (agners) [inline letsencrypt/DOCS.md:233]:
```suggestion
  <https://support.google.com/domains/answer/7630973#acme_dns>
```

**Review by agners — CHANGES_REQUESTED** [review]
To get this merged and released config.yaml needs a version bump as well as `CHANGELOG.md` a change log entry and a new version.

Btw, I guess this supersedes #3019?

**Review by seanford — APPROVED** [review]

**Comment by seanford** [comment]
How close is this to being included in a future release?  Seems like it is ready to go.

File: letsencrypt/DOCS.md (hunk 2 above)
Comment (jbms) [reply letsencrypt/DOCS.md]: Done.

**Comment by jbms** [comment]
> To get this merged and released config.yaml needs a version bump as well as `CHANGELOG.md` a change log entry and a new version.
> 
> Btw, I guess this supersedes #3019?

Done.  I didn't realize #3019 had been created --- it seems to be basically the same, just missing the `google_domains_zone` option.

**Review by agners — APPROVED** [review]

**Pull request merged**